File: src/services/Catalogo/Catalogo.API/Data/Dto/ProdutoDto.cs
﻿using System.Text.Json;

namespace Catalogo.API.Data.Dto
{
  public record ProdutoDto
  {
    public string Id { get; set; } = null!;
    public string Nome { get; set; } = null!;
    public string Descricao { get; set; } = null!;
    public string ImageUrl { get; set; } = null!;
    public decimal Preco { get; set; }
    public string UnidadeMedida { get; set; } = null!;
    public string CodigoBarras { get; set; } = null!;
    public int EstoqueAlvo { get; set; }
    public int Estoque { get; set; }
    public double Rating { get; set; }
    public int RatingCount { get; set; }
    public bool IsAtivo { get; set; }

    public override string ToString()
    {
      return JsonSerializer.Serialize(this);
    }
  }
}

File: src/mobile/lib/app/utils/dto/produtos/produto_dto.dart
class ProdutoDto {
  ProdutoDto({
    required this.id,
    required this.nome,
    required this.descricao,
    required this.imageUrl,
    required this.preco,
    required this.unidadeMedida,
    required this.codigoBarras,
    required this.estoqueAlvo,
    required this.estoque,
    required this.rating,
    required this.ratingCount,
    required this.isAtivo,
  });
  late final String id;
  late final String nome;
  late final String descricao;
  late final String imageUrl;
  late final num preco;
  late final String unidadeMedida;
  late final String codigoBarras;
  late final int estoqueAlvo;
  late final int estoque;
  late final num rating;
  late final int ratingCount;
  late final bool isAtivo;

  ProdutoDto.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    descricao = json['descricao'];
    imageUrl = json['imageUrl'];
    preco = json['preco'];
    unidadeMedida = json['unidadeMedida'];
    codigoBarras = json['codigoBarras'];
    estoqueAlvo = json['estoqueAlvo'];
    estoque = json['estoque'];
    rating = json['rating'];
    ratingCount = json['ratingCount'];
    isAtivo = json['isAtivo'];
  }
}
